src/qweb/expression_parser.ts

```diff
@@ -257,16 +257,31 @@ export function compileExprToArray(expr: string, scope: { [key: string]: QWebVar
   const tokens = tokenize(expr);
 
   let i = 0;
+  let stack = []; // to track last opening [ or {
 
   while (i < tokens.length) {
     let token = tokens[i];
     let prevToken = tokens[i - 1];
     let nextToken = tokens[i + 1];
+    let groupType = stack[stack.length - 1];
+
+    switch (token.type) {
+      case "LEFT_BRACE":
+        stack.push("{");
+        break;
+      case "LEFT_BRACKET":
+        stack.push("[");
+        break;
```

Comment:
Any reason not to use the token type directly?
```suggestion
      case "LEFT_BRACE":
      case "LEFT_BRACKET":
        stack.push(token.type);
        break;
```